server/index.js

```diff
@@ -71,6 +65,108 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+// GET ALL PRODUCTS IN SHOPPING CART
+app.get('/api/cart', (req, res, next) => {
+
+  if (!req.session.cartId) {
+    return [];
+  } else {
+    const sql = `
+      SELECT
+              "c"."cartItemId",
+              "c"."price",
+              "p"."productId",
+              "p"."image",
+              "p"."name",
+              "p"."shortDescription"
+        FROM  "cartItems" AS "c"
+        JOIN  "products" AS "p" USING ("productId")
+       WHERE  "c"."cartId" = $1
+      `;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(result => res.json(result.rows))
+      .catch(err => next(err));
+  }
+});
+
+// ADD NEW ITEM INTO SHOPPING CART
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (!productId) {
+    next(new ClientError('ProductId is a required field', 400));
+  } else if (productId <= 0) {
+    next(new ClientError('ProductId must be an integer greater than 0', 400));
+  }
+
+  const sql = `
+  SELECT  "productId", "price"
+    FROM  "products"
+   WHERE  "productId" = $1
+  `;
+
+  const params = [productId];
+
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`Cannot find product with ID ${productId}`, 400));
+      } else if (req.session.cartId) {
+        const newCartItem = { cartId: req.session.cartId, price: product.price };
+        return (newCartItem);
+
+      } else {
+        const cartSql = `
+          INSERT INTO   "carts" ("cartId", "createdAt")
+               VALUES   (default, default)
+            RETURNING   "cartId"
+        `;
+        return db.query(cartSql)
+          .then(result => {
+            const cartId = result.rows[0].cartId;
+            const newCartItem = { cartId: cartId, price: product.price };
+            return newCartItem;
+          });
+      }
+    })
+    .then(result => {
+      req.session.cartId = result.cartId;
+      const sql = `
+      INSERT INTO "cartItems"
+                  ("cartId", "productId", "price")
+      VALUES      ($1, $2, $3)
+      RETURNING   "cartItemId"
+      `;
+
+      const params = [req.session.cartId, parseInt(productId), result.price];
+      return db.query(sql, params);
+    })
+    .then(result => {
+      const newCardItemId = result.rows[0].cartItemId;
+      const sql = `
+      SELECT
+              "c"."cartItemId",
+              "c"."price",
+              "p"."productId",
+              "p"."image",
+              "p"."name",
+              "p"."shortDescription"
+        FROM  "cartItems" AS "c"
+        JOIN  "products" AS "p" USING ("productId")
+       WHERE  "c"."cartItemId" = $1
+      `;
+      const params = [newCardItemId];
+      db.query(sql, params)
```

Comment:
Be sure to `return` this `db.query()` so that if the query fails for some reason, your `.catch()` handles the error and doesn't leave the client hanging.